.github/workflows/secrets-sdk.yml

```diff
@@ -0,0 +1,275 @@
+  #           freebsd-version
+  #           yarn install
+  #           yarn build
+  #           strip -x *.node
+  #           yarn test
+  #           rm -rf node_modules
+  #           rm -rf target
+  #           rm -rf .yarn/cache
+  #     - name: Upload artifact
+  #       uses: actions/upload-artifact@v3
+  #       with:
+  #         name: bindings-freebsd
+  #         path: ${{ env.APP_NAME }}.*.node
+  #         if-no-files-found: error
+  test:
+    defaults:
+      run:
+        working-directory: packages/secrets
+    name: Test bindings on ${{ matrix.settings.target }} - node@${{ matrix.node }}
+    needs:
+      - build
+    strategy:
+      fail-fast: false
+      matrix:
+        settings:
+          - host: windows-latest
+            target: x86_64-pc-windows-msvc
+          - host: macos-latest
+            target: x86_64-apple-darwin
+          - host: ubuntu-latest
+            target: x86_64-unknown-linux-gnu
+          - host: ubuntu-latest
+            target: x86_64-unknown-linux-musl
+          - host: ubuntu-latest
+            target: aarch64-unknown-linux-gnu
+            platform: linux/arm64
+          - host: ubuntu-latest
+            target: aarch64-unknown-linux-musl
+            platform: linux/arm64
+          - host: ubuntu-latest
+            target: armv7-unknown-linux-gnueabihf
+            platform: linux/arm/v7
+        node:
+          - "16"
+          - "18"
+    runs-on: ${{ matrix.settings.host }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+          check-latest: true
+          cache: npm
+      - name: Install dependencies
+        run: npm ci --ignore-scripts
+      - name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: bindings-${{ matrix.settings.target }}
+          path: packages/secrets/src/keyring/
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+        if: ${{ matrix.settings.platform }}
+        with:
+          platforms: ${{ matrix.settings.platform }}
+      - run: docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+        if: ${{ matrix.settings.platform }}
+      - name: Test bindings
+        run: npm run test
+        if: ${{ matrix.settings.host != 'ubuntu-latest' }}
+      - name: Setup and run tests
+        uses: addnab/docker-run-action@v3
+        if: ${{ matrix.settings.host == 'ubuntu-latest' && !endsWith(matrix.settings.target, 'musl') }}
+        with:
+          image: ${{ format('node:{0}-slim', matrix.node) }}
+          options: "-v ${{ github.workspace }}:/build -w /build --cap-add=IPC_LOCK ${{ matrix.settings.platform && format('--platform={0}', matrix.settings.platform) }}"
+          run: |
+            set -e
+            apt update -y && apt install -y gnome-keyring
+            dbus-run-session -- bash packages/secrets/scripts/linux-test.sh
```

Comment:
For the purpose of being consistent with other scripts like "prebuildify.sh", and to make it easier for users to run the script locally, would it make sense to remove the `cd` command in "linux-test.sh"?

Then this command could be replaced with:
`cd packages/secrets && dbus-run-session -- bash scripts/linux-test.sh`